corgi/tasks/common.py

```diff
@@ -73,3 +75,23 @@ def _create_relations(build_ids, external_system_id, product_ref, relation_type)
         if created:
             no_of_relations += 1
     return no_of_relations
+
+
+def run_external(
+    command: list[str], *args, **kwargs
+) -> tuple[subprocess.CompletedProcess, list[str]]:
+    """Simple wrapper function to securely run an external command using the subprocess module
+    Raises a CalledProcessError by default if the command returned a non-zero exit code"""
+    logger.info(f"Running external command: {command}")
+    result = subprocess.run(command, *args, **kwargs, capture_output=True, check=True, text=True)
+    logger.info(f"Command completed with result: {result}")
+    if result.stderr:
+        # Treat warnings like errors even if the command had a successful exit code
+        raise ValueError(f"Command {command} failed: {result.stderr}")
+    output = result.stdout.split("\n")
```

Comment:
Can this just strip the trailing/leading whitespace instead of trimming lines below?

```python
output = result.stdout.strip().splitlines()
```